Userland/Applications/Piano/TrackControlsWidget.cpp

```diff
@@ -7,28 +7,46 @@
  */
 
 #include "TrackControlsWidget.h"
-#include "MainWidget.h"
 #include "ProcessorParameterWidget/ParameterWidget.h"
-#include "TrackManager.h"
-#include <LibDSP/ProcessorParameter.h>
 #include <LibGUI/BoxLayout.h>
-#include <LibGUI/Label.h>
-#include <LibGfx/Orientation.h>
+#include <LibGUI/Frame.h>
+#include <LibGUI/GroupBox.h>
+#include <LibGUI/Widget.h>
 
-TrackControlsWidget::TrackControlsWidget(TrackManager& track_manager, MainWidget& main_widget)
-    : m_track_manager(track_manager)
-    , m_main_widget(main_widget)
+TrackControlsWidget::TrackControlsWidget(WeakPtr<DSP::Track> track)
+    : m_track(move(track))
 {
-    set_layout<GUI::HorizontalBoxLayout>();
-    set_preferred_width(GUI::SpecialDimension::Grow);
-    set_fill_with_background_color(true);
+}
+
+ErrorOr<NonnullRefPtr<TrackControlsWidget>> TrackControlsWidget::try_create(WeakPtr<DSP::Track> track)
+{
+    auto widget = TRY(adopt_nonnull_ref_or_enomem(new (nothrow) TrackControlsWidget(move(track))));
+
+    TRY(widget->try_set_layout<GUI::HorizontalBoxLayout>());
+    widget->set_preferred_width(GUI::SpecialDimension::Grow);
+    widget->set_fill_with_background_color(true);
+
+    auto mastering_parameters = TRY(widget->try_add<GUI::GroupBox>());
+    TRY(mastering_parameters->try_set_layout<GUI::HorizontalBoxLayout>());
+
+    auto strong_track = widget->m_track.value();
+
+    for (auto& parameter : strong_track->track_mastering()->parameters())
+        (void)TRY(mastering_parameters->try_add<ProcessorParameterWidget>(parameter));
```

Comment:
Why do you `(void)` this? Same for L46

Reply:
The function returns the widget itself, but we don't actually need to store it anywhere. The (void) cast circumvents the nodiscard of ErrorOr (even though we will properly propagate the error, which is all that matters here).